### PR TITLE
Fix migration script to take ipip mode into account

### LINKED ISSUE #3018: Calico 1.7.0 migration template enables cross-subnet mode incorrectly
Kops version: 1.7.0-beta.1
Kubernetes version: 1.7.1

https://github.com/kubernetes/kops/blob/1.7.0-beta.1/docs/calico_cidr_migration/jobs.yaml.template#L18

Doing this without setting
```
spec:
  networking:
    calico:
      crossSubnet: true
```

And doing a `rolling-update` with `--force` means that the [Source/Destination check controller](https://github.com/kubernetes/kops/blob/master/docs/networking.md#enable-cross-subnet-mode-in-calico-aws-only) doesn't get installed and packets start disappearing.
  
This broke our staging cluster after upgrading to 1.7.1

Sadly adding `crossSubnet: true` didn't pickup the addon for k8s-ec2-srcdst either (maybe the channel didn't get updated for 1.7 for it?) I'll make a separate issue for this though.

### PR DESCRIPTION
**Pull request by ottoyiu**
Fixes #3018. Thank you @blakebarnett for the report! 

The migration manifest was assuming that everyone was using cross-subnet mode, which is a big no-no. :(
This script removes the cross-subnet line from the manifest file if `crossSubnet` is not set to be `true`.

Once merged, people will be able to pull the latest version without a release since the migration README points to master for the script.

### CODE REVIEW

File: docs/calico_cidr_migration/create_migration_manifest.sh [hunk 1]
```diff
@@ -34,7 +34,17 @@ export MIGRATION_TEMPLATE="jobs.yaml.template"
...
 export POD_CIDR="`kops get cluster $NAME -o json --full | jq .spec.kubeControllerManager.clusterCIDR --raw-output`"
+export IS_CROSS_SUBNET="`kops get cluster $NAME -o json --full | jq .spec.networking.calico.crossSubnet --raw-output`"
+
 cp ${MIGRATION_TEMPLATE} ${MIGRATION_MANIFEST}
+
+if [ "$IS_CROSS_SUBNET" = "true" ]; then
+    echo "ipip mode is set to 'cross-subnet'. Honouring in migration manifest."
+else
+    echo "ipip mode is set to 'Always'. Honouring in migration manifest."
+    sed -i "/mode: cross-subnet/d" ${MIGRATION_MANIFEST}
```
Comment (blakebarnett) [inline docs/calico_cidr_migration/create_migration_manifest.sh]: Missed this in my review, but this is missing `-e` and results in:

`sed: 1: "jobs.yaml": invalid command code j`

Comment (ottoyiu) [inline docs/calico_cidr_migration/create_migration_manifest.sh]: Are you using OSX? 

seems to be me that they are equivalent at least in Linux:

```
     sed 's/hello/world/' input.txt > output.txt
     sed -e 's/hello/world/' input.txt > output.txt
```

Comment (blakebarnett) [inline docs/calico_cidr_migration/create_migration_manifest.sh]: yes, macOS sierra

**Pull request merged**